addons/account/tests/test_account_payment.py

```diff
@@ -652,6 +652,28 @@ def test_payment_confirmation_with_bank_outstanding_account(self):
         payment.action_post()
         self.assertEqual(payment.state, 'paid')
 
+    def test_payment_state_with_unreconciliable_outstanding_account(self):
+        unreconciliable_account = self.env['account.account'].create({
+            'code': '209.01.01',
+            'name': 'Bank Account',
+            'account_type': 'asset_cash',
+            'reconcile': False,
+        })
+        self.company_data['default_journal_bank'].outbound_payment_method_line_ids[0].payment_account_id = unreconciliable_account
+        invoice = self.init_invoice(move_type='out_invoice', amounts=[10], post=True)
+
+        payment = self.env['account.payment.register'].with_context(
+            active_model='account.move',
+            active_ids=invoice.ids,
+        ).create({
+            'amount': 10.0,
+            'journal_id': self.company_data['default_journal_bank'].id,
+            'payment_method_line_id': self.company_data['default_journal_bank'].outbound_payment_method_line_ids[0].id,
+            'currency_id': self.company_data['currency'].id,
+        })._create_payments()
```

Comment:
The register payment wizard already computes a lot of default values
```suggestion
        unreconciliable_account = self.env['account.account'].create({
            'code': '209.01.01',
            'name': 'Bank Account',
            'account_type': 'asset_cash',
            'reconcile': False,
        })
        self.company_data['default_journal_bank'].outbound_payment_method_line_ids.payment_account_id = unreconciliable_account
        invoice = self.init_invoice(move_type='out_invoice', amounts=[10], post=True)

        payment = self.env['account.payment.register'].with_context(
            active_model='account.move',
            active_ids=invoice.ids,
        ).create({})._create_payments()
```

Reply:
I need to at least keep the line on `'payment_method_line_id'` for the test to work